tests/virtual_machines_test.go

```diff
@@ -317,320 +213,430 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 					Expect(err).To(HaveOccurred(), "Should fail to allocate vm because there are no free mac addresses")
 				})
 			})
-			Context("and trying to create a VM after a MAC address has just been released due to a VM deletion", func() {
-				It("[test_id:2165]should re-use the released MAC address for the creation of the new VM and not return an error", func() {
-					vm1 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", ""),
-						newInterface("br2", "")}, []kubevirtv1.Network{newNetwork("br1"), newNetwork("br2")})
-
-					err := testClient.VirtClient.Create(context.TODO(), vm1)
-					Expect(err).ToNot(HaveOccurred(), "Should succeed creating vm1")
-					mac1, err := net.ParseMAC(vm1.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
-					Expect(err).ToNot(HaveOccurred(), "Should succeed parsing vm1's first mac")
-					mac2, err := net.ParseMAC(vm1.Spec.Template.Spec.Domain.Devices.Interfaces[1].MacAddress)
-					Expect(err).ToNot(HaveOccurred(), "Should succeed parsing vm1's second mac")
-
-					vm2 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br3", "")},
-						[]kubevirtv1.Network{newNetwork("br3")})
+			Context("and testing finalizers", func() {
+				Context("When the VM is not being deleted", func() {
+					It("should have a finalizer and deletion timestamp should be zero ", func() {
+						vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br", "")},
+							[]kubevirtv1.Network{newNetwork("br")})
+						err := testClient.VirtClient.Create(context.TODO(), vm)
+						Expect(err).ToNot(HaveOccurred())
 
-					err = testClient.VirtClient.Create(context.TODO(), vm2)
-					Expect(err).ToNot(HaveOccurred(), "Should succeed creating vm2")
-					_, err = net.ParseMAC(vm2.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
-					Expect(err).ToNot(HaveOccurred(), "Should succeed parsing vm2's mac")
+						Eventually(func() bool {
+							err = testClient.VirtClient.Get(context.TODO(), client.ObjectKey{Namespace: vm.Namespace, Name: vm.Name}, vm)
+							Expect(err).ToNot(HaveOccurred())
+							if vm.ObjectMeta.DeletionTimestamp.IsZero() {
+								if len(vm.ObjectMeta.Finalizers) == 1 {
+									if strings.Compare(vm.ObjectMeta.Finalizers[0], pool_manager.RuntimeObjectFinalizerName) == 0 {
+										return true
+									}
+								}
+							}
+							return false
+						}, timeout, pollingInterval).Should(BeTrue())
+					})
+				})
+			})
+		})
+	})
+})
 
-					newVM1 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("br1", vm1.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress),
-						newInterface("br2", vm1.Spec.Template.Spec.Domain.Devices.Interfaces[1].MacAddress)}, []kubevirtv1.Network{newNetwork("br1"), newNetwork("br2")})
+func polarionTests() {
```

Comment:
We call `polarionTests` only at one place why do we need it?

Reply:
we don't really. 
Done